orderedmap.go

```diff
@@ -136,3 +136,10 @@ func (m *OrderedMap) Back() *Element {
 		Value:   element.value,
 	}
 }
+
+// Copy will return a pointer to a new instances OrderedMap that is
+// a copy of the existing one
+func (m *OrderedMap) Copy() *OrderedMap {
+	m2 := *m
```

Comment:
This is not a copy of the data, it's a copy of the points that still point to the same data. You will need to allocate a new `OrderedMap` and copy all the items across to it by iterating the existing one.

It's also a good idea in the docs to reiterate a few things:

```go
// Copy returns a new OrderedMap with the same elements.
// The elements are not deep copied.
// Using Copy while there are concurrent writes may mangle the result.
```

Reply:
I understand:

> This is not a copy of the data, it's a copy of the pointers that still point to the same data. You will need to allocate a new OrderedMap and copy all the items across to it by iterating the existing one.

But then in the docs, you have suggested:

> The elements are not deep copied.

To me, a "deep copy" means a "a copy of the data". Did you mean by "The elements are not deep copied." what I had implemented and not what I should have implemented? If that makes sense? 😄